core/api/service/api_service.cpp

```diff
@@ -186,10 +213,26 @@ namespace kagome::api {
         "Internal error. No session was bound to subscription.");
   }
 
+  outcome::result<uint32_t> ApiService::subscribeNewHeads() {
+    if (auto session_id = threaded_info.fetchSessionId(); session_id) {
+      if (auto session_context = findSessionById(*session_id)) {
+        auto &session = session_context->events_subscription;
+        const auto id = session->generateSubscriptionSetId();
+        session->subscribe(id, primitives::SubscriptionEventType::kNewHeads);
+        return static_cast<uint32_t>(id);
+      }
+      throw jsonrpc::InternalErrorFault(
+          "Internal error. No session was stored for subscription.");
+    }
+    throw jsonrpc::InternalErrorFault(
+        "Internal error. No session was bound to subscription.");
+  }
+
   outcome::result<void> ApiService::unsubscribeSessionFromIds(
       const std::vector<uint32_t> &subscription_ids) {
     if (auto session_id = threaded_info.fetchSessionId(); session_id) {
-      if (auto session = findSessionById(*session_id)) {
+      if (auto session_context = findSessionById(*session_id)) {
+        auto &session = session_context->storage_subscription;
```

Comment:
I guess, the same needed for event_subscription

Reply:
No, this unsubscribe request for ```storage_subscription```. I wonder why they did this, but in spec each unsubscribtion type have have the independent request.